src/httprpc.cpp

```diff
@@ -180,6 +188,49 @@ static bool HTTPReq_JSONRPC(HTTPRequest* req, const std::string &)
         // Set the URI
         jreq.URI = req->GetURI();
 
+#ifdef ENABLE_WALLET
+        static const std::string WALLET_ENDPOINT_BASE = "/wallet/";
```

Comment:
Can we not allow an enumeration of possible users here and then have the user->wallet mapping checked in rpcwallet.cpp?

Reply:
`rpcwallet.cpp` is already past the point where the GUI and RPC abstractions combine.

Also, we are already enumerating the possible users here. To move it would mean two enumerations...

Reply:
I was primarily echoing @jnewbery's comments above about needing cleanups, especially not introducing a bunch more ifdef ENABLE_WALLETs, and this is maybe an obvious case where you seem to be possible-needlessly moving code from src/wallet to src/httprpc. It may be a bit more effecient, but I'm not sure its worth mucking up more wallet stuff in httprpc here.

Reply:
There is no src/wallet code that is strictly for RPC. I could make a function there, I suppose, and call it from here, but that seems just plain ugly?

Reply:
Why not throw some additional registration stuff in src/wallet/rpcwallet? There's already registration stuff there now. httprpc should be, IMO, as much as possible, a "dumb dispatcher" - wallet should tell RPC what it wants, and RPC can pass in the parameters it was given by the client and rpcwallet.cpp can handle how to deal with them.